core/services/pg/lease_lock.go

```diff
@@ -126,26 +120,33 @@ func (l *leaseLock) TakeAndHold(sig shutdown.Signal) (err error) {
 		l.logRetry(retryCount)
 		retryCount++
 		select {
-		case <-l.chStop:
+		case <-ctx.Done():
 			err = errors.New("stopped")
 			if l.conn != nil {
 				err = multierr.Combine(err, l.conn.Close())
 			}
 			return err
-		case <-sig.Wait():
-			return multierr.Combine(
-				errors.New("application shutdown"),
-				l.conn.Close(),
-			)
 		case <-time.After(utils.WithJitter(l.refreshInterval)):
 		}
 	}
 	l.logger.Debug("Got exclusive lease on database")
-	l.wg.Add(1)
-	go l.loop()
+
+	lctx, cancel := context.WithCancel(context.Background())
```

Comment:
Might be worth making a comment here to explain why the passed in `ctx` is not used, and re-assuring the reader that this is deliberate and not a mistake

Reply:
Done